src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.BSD.cs

```diff
@@ -57,6 +57,8 @@ private static IntPtr ProcessorAffinityCore
         /// </summary>
         private void GetWorkingSetLimits(out IntPtr minWorkingSet, out IntPtr maxWorkingSet)
         {
+            EnsureState(State.HaveNonExitedId);
```

Comment:
This looks like a good change if you have tested it on FreeBSD, but it would be best to submit it as a separate improvement and not part of port of other platform.

Reply:
Right, though it's pretty simple.  The test is complaining that we throw the wrong exception code.
This makes it throw the right code for a process object that has not yet started.

I'm not in a position test on FreeBSD at the moment.  Are there FreeBSD people around who could help me?

Reply:
> I'm not in a position test on FreeBSD at the moment. Are there FreeBSD people around who could help me?

Which is why it is best to put it in a separate PR, so someone can test before approving it.

Reply:
Opened an issue and a PR
https://github.com/dotnet/runtime/issues/105422
https://github.com/dotnet/runtime/pull/105424

Reply:
Doing this stuff in a separate PR